easybuild/easyblocks/d/deepspeed.py

```diff
@@ -0,0 +1,61 @@
+from easybuild.easyblocks.generic.pythonpackage import PythonPackage
+from easybuild.tools.build_log import EasyBuildError
+from easybuild.tools.config import build_option
+import easybuild.tools.environment as env
+
+
+class EB_DeepSpeed(PythonPackage):
+    """Custom easyblock for DeepSpeed"""
+
+    @staticmethod
+    def extra_options():
+        """Change some defaults for easyconfig parameters."""
+        extra_vars = PythonPackage.extra_options()
+        extra_vars['use_pip'][0] = True
+        extra_vars['download_dep_fail'][0] = True
+        extra_vars['sanity_pip_check'][0] = True
+        return extra_vars
+
+    def __init__(self, *args, **kwargs):
+        """Initialize DeepSpeed easyblock."""
+        super().__init__(*args, **kwargs)
+
+        dep_names = set(dep['name'] for dep in self.cfg.dependencies())
+
+        # require that PyTorch is listed as dependency
+        if 'PyTorch' not in dep_names:
+            raise EasyBuildError('PyTorch not found as a dependency')
```

Comment:
I think we shouldn't error inside the init method, maybe move this to the build steps instead? The test suite doesn't like this at all.

Reply:
Moving it to config_step instead, but could just as well just drop it.